Changes to included genders for 0.0.3

ceceradio opened the pull request:
Added Boi, and Butch (to coincide with the inclusion of Femme). Removed "Trans Femme", changed position of "Trans Feminine" to be alphabetically accurate. Added "Cis Man" which was accidentally left out.

Please leave a comment, @ me on twitter (@satsukitv) or send an ask to satsukitv.tumblr.com if you have any feedback whatsoever.

Good faith requests to add, remove, or change genders are welcome.

Commit:
added Boi, and Butch (to coincide with the inclusion of Femme). Removed "Trans Femme", changed position of "Trans Feminine" to be alphabetically accurate. Added "Cis Man" which was accidentally left out.

satsukitv commented:
Open questions for dialogue:

- Are there any common genders missing? (Full list: [https://github.com/satsukitv/genders/blob/dev-0.0.3/genders.json](https://github.com/satsukitv/genders/blob/dev-0.0.3/genders.json))
- Similar to the genders.angular.js file, are there any "low hanging fruit" frameworks that the genders could be implemented?
- Is there a need or want for subsets of the genders? Such as a limited subset of genders without cis or trans prefixes?

Commit:
From twitter feedback:
* Add boy and girl
* Add demiboy

Additionally, added demiwoman to provide an alternative to demigirl.

satsukitv commented:
Someone on twitter pointed out that it may not be appropriate to have intersex as a gender, especially considering that intergender is available as an option. As neither of us are intersex, I'm hoping to get some feedback from the intersex community on this potential change. Is intersex appropriate in the list?

Commit:
From twitter feedback:
- Added "Gender Neutral"

Also, synchronized genders.angular.js with the json file.

Commit:
based on twitter feedback: Added ambigender

satsukitv commented:
Any thoughts of adding "Gal" as an alternative to "Girl"?

Commit:
Added femboy from twitter feedback.

ledge23 commented:
Suggestion: Add 'Two Spirit'.

jdodson commented:
This seems good, adding "Two Spirit" sounds like a fine addition.

Could this be merged?
